internal/driver/controllerserver_helper.go

```diff
@@ -95,31 +95,6 @@ type VolumeParams struct {
 	Region           string
 }
 
-// canAttach indicates whether or not another volume can be attached to the
-// Linode with the given ID.
-//
-// Whether or not another volume can be attached is based on how many instance
-// disks and block storage volumes are currently attached to the instance.
-func (cs *ControllerServer) canAttach(ctx context.Context, instance *linodego.Instance) (canAttach bool, err error) {
-	log := logger.GetLogger(ctx)
-	log.V(4).Info("Checking if volume can be attached", "instance_id", instance.ID)
-
-	// Get the maximum number of volume attachments allowed for the instance
-	limit, err := cs.maxAllowedVolumeAttachments(ctx, instance)
-	if err != nil {
-		return false, err
-	}
-
-	// List the volumes currently attached to the instance
-	volumes, err := cs.client.ListInstanceVolumes(ctx, instance.ID, nil)
-	if err != nil {
-		return false, errInternal("list instance volumes: %v", err)
-	}
-
-	// Return true if the number of attached volumes is less than the limit
-	return len(volumes) < limit, nil
-}
-
 // maxAllowedVolumeAttachments calculates the maximum number of volumes that can be attached to a Linode instance,
 // taking into account the instance's memory and currently attached disks.
 func (cs *ControllerServer) maxAllowedVolumeAttachments(ctx context.Context, instance *linodego.Instance) (int, error) {
```

Comment:
You can also remove this helper func(). Its not used anywhere else except the funcs you are removing :)